include/dlaf/gpu/blas/api.h

```diff
@@ -0,0 +1,83 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+/// @file
+
+#ifdef DLAF_WITH_HIP
+
+#include <hipblas.h>
+#include <rocblas.h>
+
+#define cublasCall           hipblasCall
+#define cublasCherk          hipblasCherk
+#define cublasCreate         hipblasCreate
+#define cublasCtrmm          hipblasCtrmm
+#define cublasCtrsm          hipblasCtrsm
+#define cublasDaxpy          hipblasDaxpy
+#define cublasDestroy        hipblasDestroy
+#define cublasDgemm          hipblasDgemm
+#define cublasDiagType_t     hipblasDiagType_t
+#define cublasDtrmm          hipblasDtrmm
+#define cublasDtrsm          hipblasDtrsm
+#define cublasFillMode_t     hipblasFillMode_t
+#define cublasGetErrorString hipblasGetErrorString
+#define cublasGetStream      hipblasGetStream
+#define cublasHandle_t       hipblasHandle_t
+#define cublasOperation_t    hipblasOperation_t
+#define cublasPointerMode_t  hipblasPointerMode_t
+#define cublasSetPointerMode hipblasSetPointerMode
+#define cublasSetStream      hipblasSetStream
+#define cublasSgemm          hipblasSgemm
+#define cublasSideMode_t     hipblasSideMode_t
+#define cublasStrmm          hipblasStrmm
+#define cublasStrsm          hipblasStrsm
+#define cublasZgemm          hipblasZgemm
+#define cublasZherk          hipblasZherk
+#define cublasZtrmm          hipblasZtrmm
+#define cublasZtrsm          hipblasZtrsm
+#define cuComplex            hipblasComplex
+#define cuDoubleComplex      hipblasDoubleComplex
+
+#define CUBLAS_DIAG_NON_UNIT     HIPBLAS_DIAG_NON_UNIT
+#define CUBLAS_DIAG_UNIT         HIPBLAS_DIAG_UNIT
+#define CUBLAS_FILL_MODE_FULL    HIPBLAS_FILL_MODE_FULL
+#define CUBLAS_FILL_MODE_LOWER   HIPBLAS_FILL_MODE_LOWER
+#define CUBLAS_FILL_MODE_UPPER   HIPBLAS_FILL_MODE_UPPER
+#define CUBLAS_OP_C              HIPBLAS_OP_C
+#define CUBLAS_OP_N              HIPBLAS_OP_N
+#define CUBLAS_OP_T              HIPBLAS_OP_T
+#define CUBLAS_POINTER_MODE_HOST HIPBLAS_POINTER_MODE_HOST
+#define CUBLAS_SIDE_LEFT         HIPBLAS_SIDE_LEFT
+#define CUBLAS_SIDE_RIGHT        HIPBLAS_SIDE_RIGHT
+
+namespace dlaf {
+
+// Error handling
+using gpublasStatus_t = hipblasStatus_t;
+constexpr gpublasStatus_t GPUBLAS_STATUS_SUCCESS = HIPBLAS_STATUS_SUCCESS;
+
+}
+
+#elif defined(DLAF_WITH_CUDA)
+
+#include <cuComplex.h>
+#include <cublas_v2.h>
+
+namespace dlaf {
+
+// Error handling
+using gpublasStatus_t = cublasStatus_t;
+constexpr gpublasStatus_t GPUBLAS_STATUS_SUCCESS = CUBLAS_STATUS_SUCCESS;
+
+}
```

Comment:
Are these also meant to be public?

Reply:
Yes.

Reply:
Ok, maybe I should've been more explicit in my question :P _why_ are these public? I would not recommend trying to provide a generic HIP/ROCm/CUDA abstraction layer (which is what these definitions are part of), at least at this point. This is far from enough to provide such a layer, and with the HIP/ROCm situation it's far too unstable to consider public in my opinion. It's also far from the core scope of DLA-Future and only required for algorithm development and internals. I understand that eventually such a layer may be useful, but as long as it's not even close to complete and more importantly not needed by any users I would make it internal.

Reply:
We cannot just distinguish between user and developer. We should consider advanced users which might want to setup matrices or post-process the result using tasks. Therefore we should not hide our basic CUDA/HIP+cublas/rocblas abstraction layer but let it accessible to users.

Reply:
I still don't see the reason why we should make them internal now and move them again later, but fine, make them internal.

Reply:
Because is this even something that DLAF needs to provide? pika is having to do the same translation as DLAF. A separate library would perhaps even be more appropriate. This is essentially doing what HIP does without pulling in HIP.